ci/docker/protochecker/docker-compose.yml

```diff
@@ -0,0 +1,50 @@
+volumes:
+  generated-repo:
+
+
+services:
+  setup:
+    build:
+      context: .
+      dockerfile: Dockerfile.setup
+    image: protochecker-setup
+    volumes:
+      - ../../..:/src:ro
+      - generated-repo:/repo
+    environment:
+      - USERID=${UID:-1000}
+      - GID=${GID:-1000}
+
+  app-protobuf-gen:
+    image: ghcr.io/nordsecurity/nordvpn-linux/generator:1.4.2
+    command: "ci/generate_protobuf.sh"
+    depends_on:
+      setup:
+        condition: service_completed_successfully
+    volumes:
+      - generated-repo:/work
+    working_dir: /work
+    user: "${UID:-1000}:${GID:-1000}"
+
+  gui-protobuf-gen:
+    image: ghcr.io/nordsecurity/nordvpn-linux/protobuf_dart-3.8.1:1.0.1
+    command: "gui/scripts/generate_protobuf.sh"
+    depends_on:
+      setup:
+        condition: service_completed_successfully
+    volumes:
+      - generated-repo:/code/app
+    working_dir: /code/app
+    user: "${UID:-1000}:${GID:-1000}"
+  protochecker:
+    image: alpine/git
+    command: [ "diff", "--quiet" ]
```

Comment:
[suggestion]
Why not use git diff here?

With `git diff --name-only` we'd be able to produce nice diagnostics here, which might be a bit more user-friendly at CI checks stage?